### ISSUE TITLE
Use of reserved javascript words as properties leads to IE errors

### ISSUE THREAD

**Issue by cmeresse** [swagger-api/swagger-ui#843]
swagger-client.js uses these javascript reserved words as properties: enum, default, in and this make IE raise errors.

For instance on line 13:
```javascript
  var requiredFields = definition.enum || [];
```
definition['enum'] should be used instead of definition.enum or enum should be renamed to something else.

**Comment by mohsen1** [comment]
Could you make a PR to fix this?

**Comment by cmeresse** [comment]
Yes, but the swagger team should chose the solution:

1. Rename the properties (for instance enum -> enumValues, default -> defaultValue, in -> location)
This may be the cleanest solution but this means breaking the api backward compatibility, modifying the spec, the examples etc... a huge work (I would not do it myself)

2. Change the way to access every properties.
I would personally not chose this solution, it's a bit risky and less readable I think

3. Change the way to access only the impacted properties.
This is the quicker and safer way but not very nice to read as it will introduce inconsistency in the code like this:
```javascript
      param.allowableValues = {};
      param.isList = true;
      param.['enum'] = ["true", "false"];
```

**Comment by mohsen1** [comment]
What version of IE is affected with this? I would suggest doing the solution 3 if it's necessary to fix.

**Comment by cmeresse** [comment]
IE 9 but maybe in compatible "IE 7 Document Mode"... not sure. I will check tomorrow when I am at work.

**Comment by webron** [comment]
@fehguy should provide input here, but I can certainly say that option 1 is... not an option ;) The spec is not going to change.

**Comment by fehguy** [comment]
I agree with @mohsen1, seems like option 3 is best.  A PR would be greatly appreciated, but swagger-client.js is in the swagger-js project, develop_2.0 branch.

**Comment by cmeresse** [comment]
I have checked and IE complains only if it is running in an older "Document Mode" (Quirks, 7 or 8).
Do I create the PR anyway ?

Also, in shred.bundle.js I noticed an extra comma on line 2210. I had done a PR but I did it in swagger-ui, should I do it in swagger-js ?

Thanks

**Comment by fehguy** [comment]
addressed in 0e69c5e
